### DISCUSSION TITLE
Suggestion for App-specific Dekstop Shortcuts

### DISCUSSION THREAD

**Discussion by YaBoiMattyK** [Genymobile/feedback#75] [Genymotion Desktop] [answered]
License: Personal Use
Context: Genymotion has been a blessing for android gaming on the steam deck, as bluestacks isn't compatible with it or linux. Due to Genymotion being built more for development, I was hoping that the gaming department could get some love for linux-based users.

One feature I would like to see is the ability to make desktop shortcuts that launch a specific app (and it's corresponding phone) so I could add these apps directly to my steam library (which would also let me add game-specific control maps, since asking for native genymotion touch-mapping would probably be a bit too much). 

I understand if this request is not feasible, and if your priorities are focused towards developer features. I at least wanted to express my interest in a feature like this.

Best regards,
MattyK

**Comment by GenyPmlulu** [comment]
Hello! 
I would love to have your feedback on how you use Genymotion on your steam deck (inputs / mappings / struggles)
Best regards,

**Reply by YaBoiMattyK** [reply]
Hi!
I’m glad you have taken interest in Steam Deck usage.

So far, I’ve only used Genymotion for android gaming. Genymotion runs well enough on the Steam Deck with games such as Plants vs Zombies 2 Reflourished and Kingdom Rush. Battery life isn’t too much of a concern in the aforementioned apps, but using a browser like Firefox within an emulated phone puts heavy strain on the deck, reducing a typical full charge duration of 5 hours (with Genymotion running) down to 1 hour 30 minutes (and in some cases, lower). That is the equivalent of running a modern AAA game, such as WWE 2k23.
I have attempted to play the Chinese port of Plants vs Zombies 2, but launching it causes instant app crashes that built up over 500 crash logs. It may have also caused some memory build-up that I cannot seem to track down, adding 6 gigabytes somewhere (the apk was from an official source).
I have also attempted to play some Abandonware games on an older android build, with varying success due to the nature of downloading .apkx abandonware. Games like Army of Darkness Defense work well in that regard since they’re .apk, and not .apkx.

When I play Genymotion undocked, I use my right trackpad as a cursor and my right trigger as a left mouse click. Very responsive, and almost feels natural with games like PvZ 2 Reflourished. When I’m using a dock, I use a controller to play Genymotion. My right stick is my cursor, and my right trigger is left click. While not as accurate as a trackpad, it’s still nice playing PvZ 2 Rfl on a Television. In both uses, my start button also pauses PvZ 2 Rfl.
In the Steam Deck’s gaming mode, the emulated phone comfortably fills the screen and television.

In steam’s game overlay, I have attempted to set additional control maps to specific buttons (Move cursor to set XY position on screen, wait after set time, activate left click, return cursor to original position). That way, I can use a button on my Deck or 3rd Party Controller to “tap” a specific button in my game immediately. Sadly, that feature on Steam’s behalf seems to be bugged, as it also happens on other steam games on the Steam Deck. I have already submitted a bug report to Valve on that regard.
If Genymotion were to ever use the touch-mapping/smart-controls feature that Bluestacks, Noxplayer, and MEmu utilize, then Genymotion gaming would feel far more natural with both the deck itself and a 3rd-party controller, as it would allow for simultaneous commands that a cursor cannot make. Seeing as how the aforementioned programs do not run on Steam Deck’s native Linux OS, or the Proton translation layer, migrating to those programs for those features is not an option.

I would still like the ability to create desktop shortcuts directly to my Android games (and their corresponding phone) so I can remove the navigation that would normally be required to play them in the Steam Deck’s gaming mode (using genymotion from within the library).

Over-all, I am quite satisfied with Genymotion. And I am thankful that an android emulator like this is available for Linux-based devices. Thank you for taking your time to listen to my feedback. I hope this information is helpful.

Best regards,
MattyK

**Reply by agavignet** [reply, marked as answer]
> I’m glad you have taken interest in Steam Deck usage.

We actually wanted to publish a specific version of Genymotion on Steam store one year ago, but that has been rejected by Valve without explanation, so we paused this integration abruptly.

Thanks for your feeback, the need you're describing makes perfect sense to us. We cant promise anything regarding a delivery but it will help us in the future to create a gaming feature pack.

Adrien
